openapi-cli/src/test/java/io/ballerina/openapi/generators/client/EnumGenerationTests.java

```diff
@@ -0,0 +1,174 @@
+                .withResourceMode(true).build();
+        BallerinaClientGenerator ballerinaClientGenerator = new BallerinaClientGenerator(oasClientConfig);
+        SyntaxTree syntaxTree = ballerinaClientGenerator.generateSyntaxTree();
+        List<Diagnostic> diagnostics = getDiagnostics(syntaxTree, openAPI, ballerinaClientGenerator);
+        Assert.assertTrue(diagnostics.isEmpty());
+        compareGeneratedSyntaxTreeWithExpectedSyntaxTree(expectedPath, syntaxTree);
+    }
+
+    @Test(description = "Tests for all the nullable enum scenarios in resource function parameter generation:" +
+            "Use case 01 : Nullable enum in query parameter" +
+            "Use case 02 : Nullable enum in path parameter" +
+            "Use case 03 : Nullable enum in header parameter" +
+            "Use case 04 : Nullable enum in reusable parameter" +
+            "Use case 05 : Nullable enum in parameter with referenced schema")
+    public void generateResourceParametersWithNullableEnums() throws IOException, BallerinaOpenApiException,
+            FormatterException {
+        Path definitionPath = RES_DIR.resolve("swagger/parameters_with_nullable_enums.yaml");
+        Path expectedPath = RES_DIR.resolve("ballerina/parameters_with_nullable_enums_resource.bal");
+        OpenAPI openAPI = GeneratorUtils.normalizeOpenAPI(definitionPath, true);
+        OASClientConfig.Builder clientMetaDataBuilder = new OASClientConfig.Builder();
+        OASClientConfig oasClientConfig = clientMetaDataBuilder
+                .withFilters(filter)
+                .withOpenAPI(openAPI)
+                .withResourceMode(true).build();
+        BallerinaClientGenerator ballerinaClientGenerator = new BallerinaClientGenerator(oasClientConfig);
+        SyntaxTree syntaxTree = ballerinaClientGenerator.generateSyntaxTree();
+        List<Diagnostic> diagnostics = getDiagnostics(syntaxTree, openAPI, ballerinaClientGenerator);
+        Assert.assertTrue(diagnostics.isEmpty());
+        compareGeneratedSyntaxTreeWithExpectedSyntaxTree(expectedPath, syntaxTree);
+    }
+
+    @Test(description = "Test unsupported nullable path parameter with enums",
+            expectedExceptions = BallerinaOpenApiException.class,
+            expectedExceptionsMessageRegExp = "Path parameter value cannot be null.")
+    public void testNestedArrayQueryParamGeneration() throws IOException, BallerinaOpenApiException {
+        OpenAPI openAPI = getOpenAPI(RES_DIR.resolve("swagger/path_param_nullable_enum.yaml"));
+        OASClientConfig.Builder clientMetaDataBuilder = new OASClientConfig.Builder();
+        OASClientConfig oasClientConfig = clientMetaDataBuilder
+                .withFilters(filter)
+                .withOpenAPI(openAPI)
+                .withResourceMode(true).build();
+        BallerinaClientGenerator ballerinaClientGenerator = new BallerinaClientGenerator(oasClientConfig);
+        SyntaxTree syntaxTree = ballerinaClientGenerator.generateSyntaxTree();
+    }
```

Comment:
Is this the only negative test scenario we have for enums? Better to recheck whether we have any remaining cases.

Reply:
When comes to enum specific usecase this would the only negative use case. In other cases when the enum is not supported the generic error for unsupported types will be thrown. `"Unsupported OAS data type`